src/controller/systemartifact/execution.go

```diff
@@ -0,0 +1,153 @@
+		execMgr:           task.ExecMgr,
+		taskMgr:           task.Mgr,
+		systemArtifactMgr: systemartifact.Mgr,
+		makeCtx:           orm.Context,
+	}
+}
+
+type controller struct {
+	execMgr           task.ExecutionManager
+	taskMgr           task.Manager
+	systemArtifactMgr systemartifact.Manager
+	makeCtx           func() context.Context
+}
+
+func (c *controller) Start(ctx context.Context, async bool, trigger string) error {
+	execId, err := c.execMgr.Create(ctx, VendorTypeSystemArtifactCleanup, 0, trigger)
+	if err != nil {
+		return err
+	}
+	// cleanup job would always be scheduled in async mode in production
+	// allowing for sync mode execution only for test mode purposes
+	// if there are any trigger settings then pass them to the cleanup manager first
+	jobParams := job.Parameters{}
+
+	if !async {
+		err := c.createCleanupTask(ctx, jobParams, execId)
+		if err != nil {
+			log.Errorf("failed to create system artifact clean-up task: %v", err)
+			return err
+		}
+
+		logger.Info("Created job for scan data export successfully")
+		return nil
+	}
+	go func(ctx context.Context) {
+		err := retry.Retry(func() error {
+			_, err := c.execMgr.Get(ctx, execId)
+			return err
+		})
+		if err != nil {
+			log.Errorf("failed to get the execution %d for the export data cleanup job", execId)
+			return
+		}
+		err = c.createCleanupTask(ctx, jobParams, execId)
+		if err != nil {
+			logger.Errorf("Encountered error in scan data artifact cleanup : %v", err)
+			return
+		}
+	}(c.makeCtx())
+
+	return nil
+}
+
+func (c *controller) createCleanupTask(ctx context.Context, jobParams job.Parameters, execId int64) error {
+	j := &task.Job{
+		Name: job.SystemArtifactCleanup,
+		Metadata: &job.Metadata{
+			JobKind: job.KindGeneric,
+		},
+		Parameters: jobParams,
+	}
+
+	_, err := c.taskMgr.Create(ctx, execId, j)
+
+	if err != nil {
+		logger.Errorf("Unable to create a scan data export job in clean-up mode : %v", err)
+		c.markError(ctx, execId, err)
+		return err
+	}
+	return nil
+}
+
+func (c *controller) markError(ctx context.Context, executionID int64, err error) {
+	logger := log.GetLogger(ctx)
+	// try to stop the execution first in case that some tasks are already created
+	if err := c.execMgr.StopAndWait(ctx, executionID, 10*time.Second); err != nil {
+		logger.Errorf("failed to stop the execution %d: %v", executionID, err)
+	}
+	if err := c.execMgr.MarkError(ctx, executionID, err.Error()); err != nil {
+		logger.Errorf("failed to mark error for the execution %d: %v", executionID, err)
+	}
+}
+
+// ScheduleCleanupTask schedules a system artifact cleanup task
+func ScheduleCleanupTask(ctx context.Context) {
+	scheduleSystemArtifactCleanJob(ctx)
+}
+
+func scheduleSystemArtifactCleanJob(ctx context.Context) {
+	cronSpec := "0 0 0 * * *"
+
+	schedule, err := getSystemArtifactCleanupSchedule(ctx)
+	if err != nil {
+		return
+	}
+	if schedule != nil {
+		logger.Infof(" Export data cleanup job already scheduled with ID : %v.", schedule.ID)
```

Comment:
set it to debug, otherwise by default, it will print on launching core.

Reply:
Done